### PR TITLE
#160845491 Run code-climate test coverage

### PR DESCRIPTION
**Pull request by collinewait**
#### What does this PR do?
* It runs code-climate test coverage after
    passing the tests from Travis.
#### Description of Tasks to be completed?
* Adding a test reporter id to Travis environment variables.
* Updating Travis file.
* Updating README file with Travis and code-climate badges. 
#### How should this be manually tested?
- running builds on Travis and observing the test coverage badge. 
#### What are the relevant pivotal tracker stories?

[#160845491](https://www.pivotaltracker.com/story/show/160845491)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,13 +4,17 @@ node_js:
...
+  - chmod +x ./cc-test-reporter
+  - ./cc-test-reporter before-build
 script:
 - npm test
 - npm run build
 - npm install coveralls
-- npm test -- -u --coverage
-after_success:
-- coveralls < coverage/lcov.info
+- npm run test -- --coverage --no-cache
```
Comment (reiosantos) [inline .travis.yml]: Why do we have `--` and `--no-cache` ?
And not `npm run test --coverage`.

Comment (collinewait) [reply .travis.yml]: To run jest via npm test, we can use the command line arguments by inserting a -- between npm test and the Jest arguments as specified in the [docs](https://deltice.github.io/jest/docs/en/cli.html).  I tried npm run test --coverage and it didn't work as expected.   For --no-cache, jest may give wrong coverage details due to caching. When caching is diabled, we won't run into wrong coverage issues.

**Comment by Estaer** [comment]
The code looks good to me, but I wonder why the test coverage badge is not showing. @collinewait Please rectify.

**Comment by collinewait** [comment]
> The code looks good to me, but I wonder why the test coverage badge is not showing. @collinewait Please rectify.

@Estaer I posted the same thing in slack. I don't know if the badge will be updated after merging as it was with coveralls. Am to search for that.

**Pull request merged**